syntax = "proto3";

package user;

option go_package = "./user";

// 用户服务定义
service UserService {
  // 用户注册
  rpc Register(RegisterRequest) returns (RegisterResponse);
  // 用户登录
  rpc Login(LoginRequest) returns (LoginResponse);
  // 获取用户信息
  rpc GetUser(GetUserRequest) returns (GetUserResponse);
  // 更新用户信息
  rpc UpdateUser(UpdateUserRequest) returns (UpdateUserResponse);
  // 验证Token
  rpc ValidateToken(ValidateTokenRequest) returns (ValidateTokenResponse);
  
  // 地址管理
  rpc AddAddress(AddAddressRequest) returns (AddAddressResponse);
  rpc GetAddresses(GetAddressesRequest) returns (GetAddressesResponse);
  rpc UpdateAddress(UpdateAddressRequest) returns (UpdateAddressResponse);
  rpc DeleteAddress(DeleteAddressRequest) returns (DeleteAddressResponse);
}

// 用户信息
message User {
  int64 id = 1;
  string username = 2;
  string email = 3;
  string phone = 4;
  string avatar = 5;
  int64 created_at = 6;
  int64 updated_at = 7;
}

// 地址信息
message Address {
  int64 id = 1;
  int64 user_id = 2;
  string name = 3;
  string phone = 4;
  string province = 5;
  string city = 6;
  string district = 7;
  string detail = 8;
  string postal_code = 9;
  bool is_default = 10;
  int64 created_at = 11;
  int64 updated_at = 12;
}

// 注册请求
message RegisterRequest {
  string username = 1;
  string email = 2;
  string password = 3;
  string phone = 4;
}

message RegisterResponse {
  bool success = 1;
  string message = 2;
  User user = 3;
  string token = 4;
}

// 登录请求
message LoginRequest {
  string email = 1;
  string password = 2;
}

message LoginResponse {
  bool success = 1;
  string message = 2;
  User user = 3;
  string token = 4;
}

// 获取用户信息请求
message GetUserRequest {
  int64 user_id = 1;
}

message GetUserResponse {
  bool success = 1;
  string message = 2;
  User user = 3;
}

// 更新用户信息请求
message UpdateUserRequest {
  int64 user_id = 1;
  string username = 2;
  string phone = 3;
  string avatar = 4;
}

message UpdateUserResponse {
  bool success = 1;
  string message = 2;
  User user = 3;
}

// 验证Token请求
message ValidateTokenRequest {
  string token = 1;
}

message ValidateTokenResponse {
  bool valid = 1;
  int64 user_id = 2;
  string message = 3;
}

// 添加地址请求
message AddAddressRequest {
  int64 user_id = 1;
  string name = 2;
  string phone = 3;
  string province = 4;
  string city = 5;
  string district = 6;
  string detail = 7;
  string postal_code = 8;
  bool is_default = 9;
}

message AddAddressResponse {
  bool success = 1;
  string message = 2;
  Address address = 3;
}

// 获取地址列表请求
message GetAddressesRequest {
  int64 user_id = 1;
}

message GetAddressesResponse {
  bool success = 1;
  string message = 2;
  repeated Address addresses = 3;
}

// 更新地址请求
message UpdateAddressRequest {
  int64 address_id = 1;
  int64 user_id = 2;
  string name = 3;
  string phone = 4;
  string province = 5;
  string city = 6;
  string district = 7;
  string detail = 8;
  string postal_code = 9;
  bool is_default = 10;
}

message UpdateAddressResponse {
  bool success = 1;
  string message = 2;
  Address address = 3;
}

// 删除地址请求
message DeleteAddressRequest {
  int64 address_id = 1;
  int64 user_id = 2;
}

message DeleteAddressResponse {
  bool success = 1;
  string message = 2;
}
